src/engine/mod.rs

```diff
@@ -4,6 +4,13 @@ use crate::{alphabet, DecodeError};
 
 pub mod fast_portable;
 
+#[cfg(all(
+    any(target_arch = "x86", target_arch = "x86_64"),
+    target_feature = "avx2",
```

Comment:
Specifying a strict `target_feature` here might prevent library users from doing dynamic cpu feature detection.

It would probably be better to tell the compiler that it's OK to generate avx2 instructions here using:
```rust
#[target_feature(enable = "avx2")]
```